### DISCUSSION TITLE
Very outdated tools in codespaces

### DISCUSSION THREAD

**Discussion by piot** [community/community#27211] [Codespaces]
<p>I can not use codespaces, since the image only contains very old and outdated versions of things that I need. For example:</p>
<pre><code class="lang-auto">clang --version
clang version 3.8.1-24 (tags/RELEASE_381/final)
</code></pre>
<p>Expected at least 11.x or 12.x Version 3.18 is from the summer of 2016 (!).</p>
<pre><code class="lang-auto">cmake --version
cmake version 3.7.2
</code></pre>
<p>Expected at least 3.17.x. Version 3.7.2 is from January 2017 (!).</p>
<p>Any help would be appreciated :slight_smile:</p>

**Comment by Chuxel** [comment, marked as answer]
<p>We’re about to update the default image to Ubuntu 20.04 which should resolve this problem.</p>
<p>However, you can also use a custom container in the mean time.  There’s a pre-built image for various Debian and Ubuntu distros with C++ tools installed.</p>
<p>Create a file called <code>.devcontainer/devcontainer.json</code> in your repository and add the following contents:</p>
<pre><code class="lang-json">{
	"name": "C++",
	"image": "mcr.microsoft.com/vscode/devcontainers/cpp:0-focal", 
	"runArgs": [ "--cap-add=SYS_PTRACE", "--security-opt", "seccomp=unconfined"],
	"settings": { 
		"terminal.integrated.shell.linux": "/bin/bash"
	},
	"extensions": [
		"ms-vscode.cpptools"
	],
	"remoteUser": "vscode"
}
</code></pre>
<p>You can find <a href="https://github.com/microsoft/vscode-dev-containers/tree/master/containers/cpp" rel="noopener nofollow ugc">more info here</a>. If you want to add more to the image, you can also reference a Dockerfile that is FROM the image referenced above.</p>
